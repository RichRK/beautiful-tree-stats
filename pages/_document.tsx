import Document, { Html, Head, Main, NextScript } from 'next/document'

class MyDocument extends Document {
  render(): JSX.Element {
    return (
      <Html>
        <Head />
        <body id="tree-stats">
          <Main />
          <NextScript />
        </body>
      </Html>
    )
  }
}

export default MyDocument
