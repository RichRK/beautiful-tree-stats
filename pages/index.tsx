import 'tailwindcss/tailwind.css'
import Head from 'next/head'
import { useEffect, useState } from 'react'
import EcologiLineChart from '../components/ecologi-line-chart'
import { MenuItem, Select, SelectChangeEvent } from '@mui/material'
import TreeDates from '../data/trees.json'
import { produceChartData } from '../services/treeService'
import TreeDate from '../models/treeDateModel'

export const Home = (): JSX.Element => {
  const [duration, setDuration] = useState('all time')
  const [treeData, setTreeData] = useState(null)

  const dateRanges = ['all time', '2021', '2020', '2019']

  useEffect(() => {
    const data = produceChartData(TreeDates)
    setTreeData(data)
  }, [])

  // useEffect(() => {
  //   console.log(treeData)
  // }, [treeData])

  const getXAxisKey = (data: TreeDate): string => {
    return data.date.toLocaleString()
  }

  const handleDurationChange = (event: SelectChangeEvent): void => {
    setDuration(event.target.value)
  }

  return (
    <div className="bg-ecologi-brown w-100 px-12 min-h-screen font-body flex flex-col items-center">
      <Head>
        <title>Create Next App</title>
        <link rel="icon" href="/favicon.ico" />
      </Head>

      <main className="py-12 max-w-full flex flex-col items-center">
        <h1 className="text-6xl text-center leading-tight m-0 mb-6">
          Welcome to Ecologi Tree Stats!
        </h1>

        <p className="text-2xl mb-10 text-center">
          Here you can find our latest tree planting numbers.
        </p>

        <Select
          className="self-end"
          disabled={!treeData}
          label="Duration"
          onChange={handleDurationChange}
          value={duration}
          variant="outlined"
        >
          {dateRanges.map((dateRange, i) => {
            return (
              <MenuItem key={i} value={dateRange}>
                {dateRange}
              </MenuItem>
            )
          })}
        </Select>

        <EcologiLineChart
          data={treeData}
          lineKey="quantity"
          xAxisKey={getXAxisKey}
        />
      </main>
    </div>
  )
}

export default Home
