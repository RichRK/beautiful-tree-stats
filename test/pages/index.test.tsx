import React from 'react'
import { screen, render } from '@testing-library/react'
import { Home } from '../../pages/index'

const page = <Home />

describe('Home page', () => {
  it('matches snapshot', () => {
    const { asFragment } = render(page, {})
    expect(asFragment()).toMatchSnapshot()
  })

  it('logo is present', () => {
    render(page)
    const image = screen.getByAltText('Ecologi') as HTMLImageElement
    expect(image.src).toContain('ecologi-logo.svg')
  })
})
