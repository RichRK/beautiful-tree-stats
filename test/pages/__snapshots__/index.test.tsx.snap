// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home page matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="bg-ecologi-brown w-100 px-12 min-h-screen font-body flex flex-col items-center"
  >
    <main
      class="pt-12 pb-24 max-w-full flex flex-col items-center"
    >
      <h1
        class="text-6xl text-center leading-tight m-0 mb-4"
      >
        Welcome to
        <img
          alt="Ecologi"
          class="inline h-24 ml-6 mr-5 pt-2"
          src="/ecologi-logo.svg"
        />
        Tree Stats!
      </h1>
      <h2
        class="text-2xl mb-10 text-center"
      >
        Here you can find our latest tree planting numbers.
      </h2>
      <div
        class="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiSelect-root self-end mb-8 css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
      >
        <div
          aria-expanded="false"
          aria-haspopup="listbox"
          class="MuiSelect-select MuiSelect-outlined MuiOutlinedInput-input MuiInputBase-input css-11u53oe-MuiSelect-select-MuiInputBase-input-MuiOutlinedInput-input"
          role="button"
          tabindex="0"
        >
          all time
        </div>
        <input
          aria-hidden="true"
          class="MuiSelect-nativeInput css-yf8vq0-MuiSelect-nativeInput"
          tabindex="-1"
          value="all time"
        />
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiSelect-icon MuiSelect-iconOutlined css-hfutr2-MuiSvgIcon-root-MuiSelect-icon"
          data-testid="ArrowDropDownIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M7 10l5 5 5-5z"
          />
        </svg>
        <fieldset
          aria-hidden="true"
          class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
        >
          <legend
            class="css-173wfxe"
          >
            <span>
              Duration
            </span>
          </legend>
        </fieldset>
      </div>
      <div
        class="recharts-responsive-container"
        style="width: 100%; height: 300px;"
      />
      <p
        class="mt-8 text-gray-600 text-sm text-center"
      >
        (Some days exceed the graph's scale, for better overall legibility.)
      </p>
      <div
        class="w-full my-16 grid grid-cols-2 gap-4 md:gap-10"
      >
        <div
          class="flex flex-col justify-center text-center shadow-md border-1 border-gray-400 p-2 sm:p-6 md:p-8 rounded-md"
        >
          <p
            class="text-4xl md:text-5xl font-bold mb-2 md:mb-3"
          >
            49,249
          </p>
          <div
            class="flex justify-center"
          >
            <h3
              class="italic"
            >
              Total trees planted
            </h3>
          </div>
        </div>
        <div
          class="flex flex-col justify-center text-center shadow-md border-1 border-gray-400 p-2 sm:p-6 md:p-8 rounded-md"
        >
          <p
            class="text-4xl md:text-5xl font-bold mb-2 md:mb-3"
          >
            14,538
          </p>
          <div
            class="flex justify-center"
          >
            <h3
              class="italic"
            >
              Most trees planted in one day
            </h3>
          </div>
        </div>
        <div
          class="flex flex-col justify-center text-center shadow-md border-1 border-gray-400 p-2 sm:p-6 md:p-8 rounded-md"
        >
          <p
            class="text-4xl md:text-5xl font-bold mb-2 md:mb-3"
          >
            1,450
          </p>
          <div
            class="flex justify-center"
          >
            <h3
              class="italic"
            >
              Known species of cacti
              <span
                class="ml-1.5 not-italic"
              >
                ðŸŒµ
              </span>
            </h3>
          </div>
        </div>
        <div
          class="flex flex-col justify-center text-center shadow-md border-1 border-gray-400 p-2 sm:p-6 md:p-8 rounded-md"
        >
          <p
            class="text-4xl md:text-5xl font-bold mb-2 md:mb-3"
          >
            35,000
          </p>
          <div
            class="flex justify-center"
          >
            <h3
              class="italic"
            >
              Estimated population of mountain zebras
              <span
                class="ml-1.5 not-italic"
              >
                ðŸ¦“
              </span>
            </h3>
          </div>
        </div>
      </div>
      <h2
        class="italic text-2xl mb-10 text-center"
      >
        Trees planted per year:
      </h2>
      <div
        class="recharts-responsive-container"
        style="width: 100%; height: 350px;"
      />
    </main>
  </div>
</DocumentFragment>
`;
